SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -313,31 +307,24 @@ void SSpatialOutputLog::FormatAndPrintRawLogLine(const FString& LogLine)
 		FormatAndPrintRawErrorLine(LogLine);
 		return;
 	}
+	FString LogCategory = LogMatcher.GetCaptureGroup(1);
+	FString LogLevelText = LogMatcher.GetCaptureGroup(2);
+	FString LogMessage = LogMatcher.GetCaptureGroup(3);
 
-	FString LogLevelText = LogMatcher.GetCaptureGroup(1);
-	FString LogMessage = LogMatcher.GetCaptureGroup(2);
-	FString LogCategory = LogMatcher.GetCaptureGroup(4);
+	const FRegexPattern LogMessagePattern = FRegexPattern(TEXT("\\[(.*)\\] (.*)"));
+	FRegexMatcher LogMatcherMessage(LogMessagePattern, LogMessage);
 
-	// For worker logs 'WorkerLogMessageHandler' we use the worker name as the category. The worker name can be found in the msg.
-	// msg=[WORKER_NAME:WORKER_TYPE] ... e.g. msg=[UnrealWorkerF5C56488482FEDC37B10E382770067E3:UnrealWorker]
-	if (LogCategory == TEXT("WorkerLogMessageHandler") || LogCategory == TEXT("Runtime"))
+	if (LogMatcherMessage.FindNext())
 	{
-		const FRegexPattern WorkerLogPattern = FRegexPattern(TEXT("\\[([^:]*):([^\\]]*)\\] (.*)"));
-		FRegexMatcher WorkerLogMatcher(WorkerLogPattern, LogMessage);
-
-		if (WorkerLogMatcher.FindNext())
-		{
-			LogCategory = WorkerLogMatcher.GetCaptureGroup(1);		  // Worker Name
-			FString WorkerType = WorkerLogMatcher.GetCaptureGroup(2); // Worker Type
-
-			if (LogCategory.StartsWith(WorkerType))
-			{
-				// We shorten the category name to make it more human readable. e.g. UnrealWorkerF5C56
-				LogCategory = LogCategory.Left(WorkerType.Len() + 5);
-			}
-
-			LogMessage = WorkerLogMatcher.GetCaptureGroup(3);
-		}
+		FString WorkerType = "UnrealWorker";
```

Comment:
This isn't correct. The worker logs can be of any type such as UnrealWorker and UnrealClient. You need to be extracting the workers name from the log.

Reply:
`[2020-10-26 11:57:38.227] [worker] [info] [UnrealWorkerDAFAD03845B91FF35C4FC990D893DE5A:UnrealWorker] UNetDriver::TickDispatch: Very long time between ticks. DeltaTime: 0.01, Realtime: 12.22. SpatialNetDriver_2` so for this log, if it were an UnrealClient how would that look like, so we know how to extract it correctly

Reply:
oh wait, it actually works now:
```
UnrealClientF20B0: Welcomed by SpatialOS (Level: /Game/Maps/Control_Small)
UnrealClientF20B0: PlayerSpawn returned from server sucessfully
UnrealWorkerF4701: Warning: GetSimulationTimeStep() - Max iterations 8 hit while remaining time 0.266745 > MaxSimulationTimeStep (0.050) for 'Character_NPC_C_18', movement 'Falling'
```

Reply:
so right now this works since we only care about the number of symbols, which is the same for UnrealWorker and UnrealClient; should we be expecting anything much longer? could change it to the first 23 symbols instead of first 17 or something like that